drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -732,20 +732,23 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		LOG_ERR("Device message is NULL");
 		return -EINVAL;
 	}
-	/* Make sure we're in a good state to start */
-	if (i2c_bus_not_available(dev)) {
-		/* reset i2c port */
-		i2c_reset(dev, I2C_RC_NO_IDLE_FOR_START);
-		/*
-		 * After resetting I2C bus, if I2C bus is not available
-		 * (No external pull-up), drop the transaction.
-		 */
+
+	if (data->i2ccs == I2C_CH_NORMAL) {
```

Comment:
Can you add a short comment indicating this condition is checking for restart

Reply:
Done

Reply:
/*
	 * If the transaction of write to read is divided into two
	 * transfers, the repeat start transfer uses this flag to
	 * exclude checking bus busy.
	 */

Reply:
Done